### ISSUE TITLE
Wrong Json Parsing

### ISSUE THREAD

**Issue by hogmuzzle** [smallrye/smallrye-graphql#957]
The logic in io.smallrye.graphql.client.typesafe.impl.json.buildValue is wrong.
if(jsonFieldValue) is null you check if the field in java model is primitive or NotNull annotated but it has no meaning. According to a graphql scheema a field must be not null if **queried**
which is a server-side responsibility. A client can query for a field or not.

**Comment by t1** [comment]
The typesafe client builds a query with all fields declared in the client POJOs. And it assumes the type of a field to be non-null if it's annotated to be so or it's primitive. The service can have a different idea of the nullability of a field and still return `null`. In such a situation, it's better to get a specific exception with a helpful message instead of a generic NPE when setting the primitive in guts of the client lib or later when accessing the presumably non-null field in the client code.

Should this behavior be documented better?

(cool user alias, BTW!)

**Comment by hogmuzzle** [comment]
I agree when a service returns null for a non-null field in this situation to get a specific exception is ok. I get a schema.graphql file and generate Java POJOs for my client. This POJOs have the whole schema but I can query for specific fields and in this case the non-null check of the fields never queried throws exceptions I do not need. The check has to be in this way. If a jsonobject  has a field and it is null but the field in POJO is primitive or annotated then an exception is thrown.

**Comment by t1** [comment]
Hmmmm.... if a schema declares a field to be non-null it has to be non-null in _all_ cases. I'd say this is a problem of the schema.

How do you generate the client?

**Comment by hogmuzzle** [comment]
I do not agree. In GQL a field should not be null if a server declares it as non-null but from a client side this field has any meaning if it was queried which is not compulsory in GQL. Of course I can make several POJO for different use cases but this approach is inconvenient. I use io.github.kobylynskyi codegen plugin.

**Comment by t1** [comment]
I took a brief look at `graphql-java-codegen` and I don't understand how you use it to generate a SmallRye Typesafe Client code, nor even why you do so. Is it possible to share your code or maybe a reproducer?

**Comment by t1** [comment]
@hogmuzzle should we close this issue?
